### ISSUE TITLE
systemd-analyze  verify failed

### ISSUE THREAD

**Issue by Mingli-Yu** [systemd/systemd#18252]
**systemd version the issue has been seen with**
 >  systemd-247.2

**Used distribution**
 > A customised embedded linux based on yocto 

**Linux kernel version used** (`uname -a`)
 > 5.4.87

**CPU architecture issue was seen on**
 > …

**Expected behaviour you didn't see**
 > …

**Unexpected behaviour you saw**
 >  \# systemd-analyze --man=false verify /lib/systemd/system/initrd-cleanup.service
initrd-cleanup.service: Command systemctl is not executable: No such file or directory
  \# cat /lib/systemd/system/initrd-cleanup.service
 \#  SPDX-License-Identifier: LGPL-2.1-or-later
 \#
 \#  This file is part of systemd.
 \#
 \#  systemd is free software; you can redistribute it and/or modify it
 \#  under the terms of the GNU Lesser General Public License as published by
 \#  the Free Software Foundation; either version 2.1 of the License, or
 \#  (at your option) any later version.

[Unit]
Description=Cleaning Up and Shutting Down Daemons
DefaultDependencies=no
AssertPathExists=/etc/initrd-release
OnFailure=emergency.target
OnFailureJobMode=replace-irreversibly
After=initrd-root-fs.target initrd-fs.target initrd.target

[Service]
Type=oneshot
ExecStart=systemctl --no-block isolate initrd-switch-root.target

**Steps to reproduce the problem**
 > …

**Comment by poettering** [comment]
So, is systemctl actually installed in $PATH? Where precisely? Does your distro play any games with paths?

Is this reproducible on a less hacky distro? i.e. fedora, debian, ubuntu or so?

At least on my system it is not.

This smells a lot like a broken local install, where systemctl wasn't properly installed.

**Comment by yuwata** [comment]
Hmm, I cannot reproduce this with `initrd-cleanup.service`, but networkd with the following drop-in fails:
```
# /etc/systemd/system/systemd-networkd.service.d/override.conf
[Service]
ExecStart=
ExecStart=valgrind --leak-check=full /var/build/systemd-networkd
ExecReload=
ExecReload=/var/build/networkctl reload
MemoryDenyWriteExecute=no
PrivateTmp=yes
Environment=SYSTEMD_LOG_LEVEL=debug SYSTEMD_MEMPOOL=0
Restart=no
```
```
$ systemd-analyze verify systemd-networkd.service
/usr/lib/systemd/system/smartd.service:10: Standard output type syslog is obsolete, automatically updating to journal. Please update your unit file, and consider removing the setting altogether.
/usr/lib/systemd/system/mcelog.service:8: Standard output type syslog is obsolete, automatically updating to journal. Please update your unit file, and consider removing the setting altogether.
/usr/lib/systemd/system/alsa-restore.service:15: Standard output type syslog is obsolete, automatically updating to journal. Please update your unit file, and consider removing the setting altogether.
systemd-networkd.service: Command valgrind is not executable: No such file or directory
```
```
$ which valgrind
/usr/bin/valgrind
```

**Comment by yuwata** [comment]
@Mingli-Yu Could you show the result of `systemctl cat initrd-cleanup.service`?

**Comment by yuwata** [comment]
Ah, if a command is a relative, then it fails.

**Comment by yuwata** [comment]
Fix is waiting in #18300.
